packages/vendure-plugin-customer-managed-groups/src/customer-managed-groups.service.ts

```diff
@@ -364,4 +371,141 @@ export class CustomerManagedGroupsService {
       isGroupAdministrator,
     };
   }
+
+  async updateGroupMember(
+    ctx: RequestContext,
+    input: UpdateCustomerManagedGroupMemberInput
+  ): Promise<CustomerManagedGroup> {
+    if (
+      !input.title &&
+      !input.firstName &&
+      !input.lastName &&
+      !input.emailAddress
+    ) {
+      throw new UserInputError(`empty body`);
+    }
+    const myGroup = await this.myCustomerManagedGroup(ctx);
+    if (myGroup) {
+      const groupAdmin = myGroup.customers.find((c) => c.isGroupAdministrator);
+      if (!groupAdmin) {
+        throw new UserInputError('group has no admin');
+      }
+      const groupAdminCustomer = await this.customerService.findOne(
+        ctx,
+        groupAdmin?.customerId,
+        ['user']
+      );
+      for (let member of myGroup.customers) {
```

Comment:
We will only update one member, so we can use `.find` here:

```ts
const customerToUpdate = myGroup.customers.find(c => c.customerId === input.customerId);
if (!customerToUpdate) {
  // Again handle failing cases first to prevent too much nesting
        throw new UserInputError(
        `Customer with id ${customerId} is not part of this customer managed group`
      );
}
```